src/bgw/job.c

```diff
@@ -121,6 +121,9 @@ bgw_job_get_all(size_t alloc_size, MemoryContext mctx)
 
 	num_tuples = scanner_scan(&scanctx);
 	Assert(list_length(list_data.list) == num_tuples);
+#ifndef TS_DEBUG
```

Comment:
This should probably be `USE_ASSERT_CHECKING`

Reply:
Otherwise I would just ignore the return value from `scanner_scan` to get rid of this. It doesn't seem to be a very useful `Assert` to be honest.

Reply:
Yeah, I'm in the camp of removing the variable. You get the length of the list from the `list_data.list` var, so the `Assert` is more or less checking to making sure `scanner` is implemented correctly (and that should be tested /enforced elsewhere).